kernel/shared/src/main/scala/cats/effect/kernel/Resource.scala

```diff
@@ -373,7 +374,8 @@ sealed abstract class Resource[F[_], +A] {
    * release actions (like the `before` and `after` methods of many test frameworks), or complex
    * library code that needs to modify or move the finalizer for an existing resource.
    */
-  def allocated[B >: A](implicit F: MonadCancel[F, Throwable]): F[(B, F[Unit])] = {
+  def allocatedCase[B >: A](
```

Comment:
```suggestion
  private[effect] def allocatedCase[B >: A](
```

@vasilmkd just want to point out that regardless of our release strategy for patches this time around, we can get this PR in like this by keeping the new API internal since it still fixes a lot of bugs in existing public APIs. Then when we are ready we can un-privatize it :)

Reply:
I think we should add it in 3.3.1, but let's wait for Daniel.

Reply:
@djspiewak 👍 in discord
https://discord.com/channels/632277896739946517/839263556754472990/922206112730722325